azure-pipelines.yml

```diff
@@ -1,67 +1,112 @@
 trigger:
-- master
-jobs:
-- job: Windows
-  pool:
-    name: Hosted VS2017
-    demands: npm
-  steps:
+  branches:
+    include:
+    - master
+    - develop
+  tags:
+    include:
+    - v*
+
+strategy:
+  matrix:
+    linux:
+      imageName: "ubuntu-16.04"
+    mac:
+      imageName: "macos-10.13"
+    windows:
+      imageName: "vs2017-win2016"
+
+pool:
+  vmImage: $(imageName)
+  demands: npm
+
+steps:
+
+  - task: UseNode@1
+    displayName: 'Install node version >=12.0.0'
+    inputs:
+      version: '>=12.0.0'
+
   - task: Npm@1
     displayName: 'Install dependencies'
     inputs:
       verbose: false
+
+  - script: node node_modules/gts/build/src/cli.js check
+    displayName: 'check gts error'
+    failOnStderr: True
+
   - task: Npm@1
     displayName: 'Compile sources'
     inputs:
       command: custom
       verbose: false
       customCommand: 'run compile'
+
+  - bash: |
+      is_prod_tag=^refs\/tags\/v?[0-9]+\.[0-9]+\.[0-9]+$
+      is_test_tag=^refs\/tags\/v?[0-9]+\.[0-9]+\.[0-9]+-[rR][cC]
+
+      # Update marketplace token
+      [[ $BUILD_SOURCEBRANCH =~ $is_prod_tag ]] && is_prod=true && marketplace_token=$PROD_MARKETPLACE_TOKEN
+      [[ $BUILD_SOURCEBRANCH =~ $is_test_tag ]] && is_test=true && marketplace_token=$TEST_MARKETPLACE_TOKEN
+      [[ "$BUILD_REASON" == "Schedule" ]] && marketplace_token=$TEST_MARKETPLACE_TOKEN
+
+      [[ $BUILD_SOURCEBRANCH =~ $is_prod_tag || $BUILD_SOURCEBRANCH =~ $is_test_tag || "$BUILD_REASON" == "Schedule" ]] && deploy_to_marketplace=true
+
+      # Set job variable
+      echo "##vso[task.setvariable variable=is_prod]$is_prod"
+      echo "##vso[task.setvariable variable=is_test]$is_test"
+      echo "##vso[task.setvariable variable=marketplace_token]$marketplace_token"
+      echo "##vso[task.setvariable variable=deploy_to_marketplace]$deploy_to_marketplace"
+    env:
+      # Pass secret variable
+      PROD_MARKETPLACE_TOKEN: $(vsciot_marketplace_token)
+      TEST_MARKETPLACE_TOKEN: $(iotdevexbuild_marketplace_token)
+    displayName: 'Define runtime variable'
+
+  - script: sudo npm install -g vsce
```

Comment:
you can try if need sudo on Linux and macOS

Reply:
Fixed.